clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaApplicationClientImpl.java

```diff
@@ -125,7 +125,8 @@ public KafkaApplicationClientImpl(
                 closeHandler);
     }
 
-    private Future<MessageConsumer> createKafkaBasedDownstreamMessageConsumer(
+    // visible for testing
+    Future<MessageConsumer> createKafkaBasedDownstreamMessageConsumer(
```

Comment:
why can't you use one of the public factory methods?

Reply:
I need to use a `MockConsumer` instead of a `KafkaConsumer` for the tests. Whenever the above method is being called, I answer that call using a mocked method which uses a `MockConsumer` to create a `MessageConsumer` as in the below code. For that purpose I also have to create a spy of `KafkaApplicationClientImpl`. 

```
doAnswer(x -> KafkaBasedDownstreamMessageConsumer
        .create(tenantId, type, kafkaConsumer, consumerConfig, msgHandler, closeHandler))
                .when(client)
                .createKafkaBasedDownstreamMessageConsumer(tenantId, type, msgHandler, closeHandler);
```

Reply:
I have changed the tests so that `spy(...)` is not used anymore. Would you mind taking a look again?